sql/plan/union.go

```diff
@@ -21,13 +21,20 @@ import (
 	"github.com/dolthub/go-mysql-server/sql"
 )
 
+const (
+	UnionType = iota
+	IntersectType
+	ExceptType
+)
+
 // Union is a node that returns everything in Left and then everything in Right
 type Union struct {
```

Comment:
- we'll need something more generic to represent the collection of these. `SetArithmeticOp`? `SetArithmeticType[Union,Intersect,Set]`
-  i am also a bit worried about confusing some of the RECURSIVE CTE rules. for example, should `WITH RECURSIVE CTE UNION ... INTERSECT ... EXCEPT ... UNION` treat all of those nodes equivalently?